pkg/controller/common/reconciler/reconciler.go

```diff
@@ -119,8 +122,8 @@ func ReconcileResource(params Params) error {
 	}
 
 	if params.NeedsRecreate != nil && params.NeedsRecreate() {
-		log.Info("Resource cannot be updated, hence will be deleted and then recreated", "kind", kind, "namespace", namespace, "name", name)
-		log.Info("Deleting resource", "kind", kind, "namespace", namespace, "name", name)
+		log.Info("Resource cannot be updated, hence will be deleted and then recreated")
+		log.Info("Deleting resource")
```

Comment:
Not from your PR but I think we could combine these two log in one:
```suggestion
		log.Info("Deleting resource as it cannot be updated, it will be recreated")
```